shell/platform/linux/fl_key_event_plugin.cc

```diff
@@ -56,15 +169,138 @@ FlKeyEventPlugin* fl_key_event_plugin_new(FlBinaryMessenger* messenger,
       messenger, channel_name == nullptr ? kChannelName : channel_name,
       FL_MESSAGE_CODEC(codec));
   self->response_callback = response_callback;
+  // Add a weak pointer so we know if the text input plugin goes away.
+  g_object_add_weak_pointer(
+      G_OBJECT(text_input_plugin),
+      reinterpret_cast<gpointer*>(&(self->text_input_plugin)));
+  self->text_input_plugin = text_input_plugin;
 
+  self->pending_events = g_ptr_array_new_with_free_func(g_object_unref);
   return self;
 }
 
-void fl_key_event_plugin_send_key_event(FlKeyEventPlugin* self,
+uint64_t fl_get_event_id(GdkEventKey* event) {
```

Comment:
Add `static` (private function).

Reply:
Multiple cases in this file. You can also remove the `fl_` prefix if this is a private function.

Reply:
Also missing comments before each function.

Reply:
I placed the private symbols into an anonymous namespace, but I can remove if that's not "normal".

Reply:
I haven't done that in the existing code, but I'm not sure what the Flutter coding standards would say on that.

Reply:
Well, the style guide for Google (which I think is what we mostly adhere to in the engine C++ code) encourages private functions to be in anonymous namespaces, but also says they can be declared static. I've been on projects (Chrome) where all private functions must be in anonymous namespaces, but I'm thinking that the C-ness of the GObject APIs probably means preferring static over anonymous namespaces, so I think I'll just leave them as static and remove the namespaces.

https://google.github.io/styleguide/cppguide.html#Unnamed_Namespaces_and_Static_Variables